### PR TITLE
Documentation on row functions

### PR DESCRIPTION
**Pull request by samukweku**
This is documentation for the available row functions in datatable, with examples.

### CODE REVIEW

File: docs/manual/row_functions.rst [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    # compare the first integer column with the rest,
+    # use rowall to find rows where all is True
+    # and filter with the resulting boolean
+    df[dt.rowall(f[1]==f[1:]), :]  
+    
+        Name	A1	A2	A3	A4
+    0	deff	0	0	0	0
+    1	def2	0	0	0	0
+    2	def4	0	0	0	0
+
```
Comment (st-pasha) [inline docs/manual/row_functions.rst:104]: How about an example for checking whether values in each row are increasing?
```
df[rowall(f[1:] >= f[:-1]), :]
```

File: docs/manual/row_functions.rst [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    df
+
+	Value1	Value2	Value3	Value4	max_min
+	    5	    4	    3	    2	    3
+	    4	    3	    2	    1	    3
+	    3	    3	    5	    1	    4
+
+Rowsum, Rowmean, Rowcount, Rowsd
+--------------------------------
+``Rowsum`` and ``Rowmean`` get the sum and mean of rows respectively; ``Rowcount`` counts the number of non-missing values in a row, while ``Rowsd`` aggregates a row to get the standard deviation
```
Comment (st-pasha) [inline docs/manual/row_functions.rst:214]: All function names are in the lower case: ``rowsum``, ``rowmean``, etc.

**Comment by st-pasha** [comment]
Thanks Samuel, this looks very nice!

**Comment by st-pasha** [comment]
This is now available on our documentation website: https://datatable.readthedocs.io/en/latest/manual/row_functions.html

**Comment by samukweku** [comment]
@st-pasha @oleksiyskononenko you guys are doing an awesome job. Really appreciate your efforts.

**Pull request merged**